File: Pig Latin Example/pigme.rb
def to_piglatin(string)
  string.gsub(/\b(\w)(\S+)\b/, '\2\1ay')
end

def capitalize(string)
  lowercase_string = string.downcase
  return string if string === lowercase_string
  lowercase_string.capitalize
end

phrase = ARGV[0];
puts "English Phrase: #{phrase}"
word_array = to_piglatin(String(phrase)).split(' ')
final_array = []
word_array.each do |string|
  final_array.push(capitalize(string))
end
puts "Pig Latin: #{final_array.join(' ')}"

File: Pig Latin Example/pigme.js
'use strict';

var finalArray = [];
var phrase = '';
var wordArray = [];

function toPigLatin(str) {
  return str.replace(/\b(\w)(\S+)\b/g, '$2$1ay');
}

function capitalize(str) {
  if ((str !== str.toLowerCase())) {
    str = str.toLowerCase();
    str = str.replace(/\b\w/, function(letter) {return letter.toUpperCase();});
  }
  return str;
}

phrase = process.argv.slice(2);
console.log("English Phrase: " + phrase);
wordArray = toPigLatin(String(phrase)).split(' ');
wordArray.forEach(function(str) {
  finalArray.push(capitalize(str));
});
console.log("Pig Latin: " + finalArray.join(' '));